### PR TITLE
feature: support to set propagation of global transaction

### PR DESCRIPTION
**Pull request by CharmingRabbit**
### Ⅰ. Describe what this PR did
Supports 4 propagations of transaction:
(1) REQUIRED: Use existing transaction context or if not, begin a new transaction  (the default propagation)
(2) REQUIRES_NEW: Begin a new transaction context.
(3) NOT_SUPPORTED: exclude the existing transaction context.
(4) SUPPORTS: If there is a transaction, join the transaction; if there is no transaction, execute it in a non transactional way

### Ⅱ. Does this pull request fix one issue?
fixed [#1515](https://github.com/seata/seata/issues/1515)

### CODE REVIEW

File: tm/src/main/java/io/seata/tm/api/Propagation.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    Propagation(String name) {
+        this.name = name;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public static Propagation get(String name) {
+        for (Propagation propagation : Propagation.class.getEnumConstants()) {
```
Comment (zjinlei) [inline tm/src/main/java/io/seata/tm/api/Propagation.java]: Propagation.values() is it enough?

File: tm/src/main/java/io/seata/tm/api/Propagation.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+ *  limitations under the License.
+ */
+package io.seata.tm.api;
+
+/**
+ * Propagation level of global transactions.
+ *
+ * @author haozhibei
+ */
+public enum Propagation {
```
Comment (zjinlei) [inline tm/src/main/java/io/seata/tm/api/Propagation.java]: Propagation is a property of the transaction, can be at the same directory as TransactionInfo.java.

File: tm/src/main/java/io/seata/tm/api/Propagation.java (hunk 1 above)
Comment (CharmingRabbit) [reply tm/src/main/java/io/seata/tm/api/Propagation.java]: I've  simplified the propagation to a pure enum class

File: tm/src/main/java/io/seata/tm/api/Propagation.java (hunk 2 above)
Comment (CharmingRabbit) [reply tm/src/main/java/io/seata/tm/api/Propagation.java]: ACK

**Pull request merged**